internal/statusapi/statusapi.go

```diff
@@ -0,0 +1,107 @@
+package statusapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/clients"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/configuration"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/proxy"
+	"github.com/julienschmidt/httprouter"
+	log "github.com/sirupsen/logrus"
+)
+
+//JenkinsStatusAPI contains API to check whether Jenkins for the current user is idle|running|starting
+type JenkinsStatusAPI interface {
+	Status(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+}
+
+// JenkinsStatusAPIImpl implements JenkinsStatusAPI
+type jenkinsStatusAPIImpl struct {
+	tenant *clients.Tenant
+	idler  clients.IdlerService
+}
+
+// JenkinsStatus represents whether jenkins is running|starting|idle
+type JenkinsStatus struct {
+	Status string `json:"status"`
+}
+
+// NewJenkinsStatusAPI creates a new instance of JenkinsStatusAPI
+func NewJenkinsStatusAPI(tenant *clients.Tenant, idler clients.IdlerService) JenkinsStatusAPI {
+	return &jenkinsStatusAPIImpl{
+		tenant: tenant,
+		idler:  idler,
+	}
+}
+
+// Status returns the Jenkins status for the current user
+func (api *jenkinsStatusAPIImpl) Status(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	log.Infof("Status API ran")
+	tokenJSON, ok := r.URL.Query()["token_json"]
```

Comment:
this should be a bearer token in a header instead of a query parameter?

Reply:
url?token_json is easy to use compared to putting it in the authorization header. Auth service uses it the same way. When you login to osio you will see openshift.io?token_json=$TOKEN_JSON.

Reply:
that is the redirection part which then requires that token to be preserved somehow so that UI can use the token while making ajax calls. If you look at the UI code, you will find that they are using Auth header instead of query param ?token_json. Bottom line is that it is easier to set a header than manipulate the query.

Reply:
@kishansagathiya see: https://github.com/fabric8-ui/fabric8-ui/pull/2895/files look for `headers.set('Authorization', 'Bearer ' + token);`

Reply:
Noted. Will do

Reply:
Done with https://github.com/fabric8-services/fabric8-jenkins-proxy/pull/268/commits/3f60200b263f94df8aecbc9448fc64e38bc11b6e